### ISSUE TITLE
[False negative] No diagnostic reported on method calls with unspecified optional `IFormatProvider`

### ISSUE THREAD

**Issue by Joe4evr** [dotnet/roslyn-analyzers#5843]
### Analyzer

**Diagnostic ID**: [CA1305](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1305): `Specify IFormatProvider` (possibly also [CA1304](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1304): `Specify CultureInfo`)

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.402](https://dotnet.microsoft.com/download/dotnet/5.0)/[SDK 6.0.200](https://dotnet.microsoft.com/download/dotnet/6.0)

### Describe the bug

CA1304/CA1305 exist to warn the developer they're calling a method not utilizing a different overload that takes a `CultureInfo`/`IFormatProvider`, so the called overload *may* have unpredictable behavior under different locales.

A problem arises when calling methods where the `IFormatProvider` is an optional parameter and therefor there is no other overload to call, but a diagnostic should still get reported if no value is specified for the optional. This primarily concerns the `Parse` overloads introduced on the numeric types taking in a `ReadOnlySpan<char>`. All these have the signature:
```cs
public static [type] Parse(ReadOnlySpan<char> s, NumberStyles style = NumberStyles.Number, IFormatProvider? provider = null)
```

### Steps To Reproduce

```cs
CultureInfo.CurrentCulture = new CultureInfo("nl-NL"); // or any other culture where the decimal separator is not '.'
ReadOnlySpan<char> span = "6.8";
var d = Decimal.Parse(span); // oops! d is actually 68 now
Console.WriteLine(d == 6.8m); // prints false
```
Alternatively, checkout this reproduction project: https://github.com/Joe4evr/AnalyzerFailRepro

### Expected behavior

Diagnostic on line 3 (line 14 in the repro project) even if it has to be a new ID, (tho I don't see very much reason to introduce a new ID for this case).

### Actual behavior

No diagnostic reported.

## Additional context

* I haven't actually checked for a repro with an optional `CultureInfo`, but I'm assuming 1304's heuristics are near-identical, so it probably has the same deal.
* I initially thought there may be a limitation in the compiler API, but at least one person in a Roslyn support channel said that it should certainly be detectable whether or not an optional parameter is user-specified at call-sites.

**Comment by Joe4evr** [comment]
@sharwell @Evangelink @jmarolf Apologies for pinging, but it's been more than 3 months and I really want some eyeballs on this.

**Comment by Evangelink** [comment]
Hey @Joe4evr! Sorry about that, we are all busy.

What you describe makes perfect sense to me. I will give it a go today.
